### PR TITLE
Added JWK and JWKS support

### PR DESCRIPTION
**Pull request by valeriomazzeo**
This pull request adds support for JWK and JWKS.
It fixes https://github.com/vapor/jwt/issues/87

Note: only RSA keys have been implemented so far.

### CODE REVIEW

**Comment by dehlen** [comment]
Is there anything I can do this get this PR merged? I would love to see this integrated, however the PR is open already half a year.

**Comment by valeriomazzeo** [comment]
@dehlen you can use https://github.com/asensei/vapor-auth-jwt

File: Tests/JWTTests/JWKTests.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import XCTest
+@testable import JWT
+
+class JWKTests: XCTestCase {
+
+    static let allTests = [
```
Comment (tanner0101) [inline Tests/JWTTests/JWKTests.swift]: Instead of supplying an `allTests` array, just run `swift test --generate-linuxmain` before committing.

**Comment by dehlen** [comment]
Great to see things coming along. I am currently using a fork for this to work and love to be able to use the stable version here with the next release. Again I am not the author of the pull request but if there is anything I can help with...

**Comment by vzsg** [comment]
@tanner0101 as this PR was for 3.0, why not merge it there and release it as 3.1.0?

**Comment by tanner0101** [comment]
@vzsg we definitely could. It's currently targeted at master which is 4.0, but I'd be happy to merge into branch [`3`](https://github.com/vapor/jwt-kit/tree/3) as well. Of course, if we do merge into 3.0, we should also add it to 4.0 at the same time. 

Looking at Swift at least, they usually merge things into master (unstable), then cherry pick to the respective versions they would like to support.

**Comment by vzsg** [comment]
Well, back in November 2018 when this PR was opened, master was still 3.0 :)
But I'm happy with this approach.

**Comment by valeriomazzeo** [comment]
@tanner0101 be careful when merging into `master` or `3`. This commit https://github.com/vapor/jwt-kit/pull/97/commits/fb65262d94ddcba4cd9e3c6cc6729e4fa19bf83a has removed the additional tests execution.

**Comment by grosch** [comment]
@tanner0101 is this going to be merged into master soon?

**Comment by amkomatz** [comment]
@tanner0101 Any update on this PR? Starting development on a major application, and we could really use this :)

**Comment by valeriomazzeo** [comment]
@amkomatz the content of this pull request is included in https://github.com/asensei/vapor-auth-jwt

We've been using this since Vapor 2.

**Comment by tanner0101** [comment]
Merging this into `3-jwks` branch so that we can have a local branch to test a merge into `3`

**Comment by tanner0101** [comment]
Update on the situation here: https://github.com/vapor/jwt-kit/pull/105#issuecomment-545219882

**Comment by valeriomazzeo** [comment]
I have reverted that `master` merge. This is the PR as it was originally intended for Vapor 3 https://github.com/vapor/jwt-kit/pull/106 .

Just as a reminder I had already implemented this for Vapor 2, but for whatever reason it wasn't ported to Vapor 3.

It would be nice if you could please incorporate this feature also in the `master` branch, for Vapor 4. Otherwise I will have to re-implement it once again :(

**Comment by tanner0101** [comment]
Added in JWT 3.1.0, thanks @valeriomazzeo! https://github.com/vapor/jwt-kit/releases/tag/3.1.0

**Comment by tanner0101** [comment]
JWK added to master branch here: https://github.com/vapor/jwt-kit/pull/107

**Pull request merged**